src/charm.py

```diff
@@ -93,36 +87,42 @@ def _on_leader_elected(self, _) -> None:
         data.setdefault("replication-password", self._new_password())
 
     def _on_start(self, event) -> None:
-        password = self._get_postgres_password()
-        # If the leader was elected and it generated a superuser password for the all the units,
-        # the cluster can be bootstrapped in each unit.
-        if password is not None:
-            try:
-                self._cluster.bootstrap_cluster(password)
-            except ClusterAlreadyRunningError:
-                logging.error("there is already a running cluster")
-                self.unit.status = BlockedStatus("there is already a running cluster")
-            except ClusterCreateError as e:
-                logging.error("failed to create cluster")
-                self.unit.status = BlockedStatus(f"failed to create cluster with error {e}")
-            except (ClusterNotRunningError, ClusterStartError) as e:
-                logging.error("failed to start cluster")
-                self.unit.status = BlockedStatus(f"failed to start cluster with error {e}")
-            except subprocess.CalledProcessError as e:
-                logging.error("failed to bootstrap cluster")
-                self.unit.status = BlockedStatus(f"failed to bootstrap cluster with error {e}")
-            else:
+        """Handle the start event."""
+        # Doesn't try to bootstrap the cluster if it's in a blocked state
+        # caused, for example, because a failed installation of packages.
+        if self._has_blocked_status:
+            return
+
+        postgres_password = self._get_postgres_password()
+        replication_password = self._get_postgres_password()
+        # If the leader was elected and it generated the needed passwords,
+        # the cluster can be bootstrapped.
+        if postgres_password is not None and replication_password is not None:
```

Comment:
nit: i usually find it more readable and less complex to avoid if/else blocks. if i were to suggest a refactor of this condition, it would be like:
```
if not postgres_password or not replication_password:
     defer()
     return

<all other logic in the if statement without an indentation - thus reduced indentation complexity>
```

Reply:
It looks a lot better this way. Thanks Shayan! Updated on https://github.com/canonical/postgresql-operator/pull/8/commits/d2d0915c789894145f29c296de6b4ec5492cf382.